nemo_text_processing/text_normalization/en/taggers/date.py

```diff
@@ -161,20 +160,36 @@ def __init__(self, cardinal: GraphFst, deterministic: bool):
         month_graph = pynutil.insert("month: \"") + month_graph + pynutil.insert("\"")
         month_numbers_graph = pynutil.insert("month: \"") + month_numbers_graph + pynutil.insert("\"")
 
+        endings = ["rd", "th", "st", "nd"]
+        endings += [x.upper() for x in endings]
+        endings = pynini.union(*endings)
+
         day_graph = (
             pynutil.insert("day: \"")
-            + ((pynini.union("1", "2", "3") + NEMO_DIGIT) | NEMO_DIGIT) @ cardinal_graph
+            + (
+                ((pynini.union("1", "2", "3") + NEMO_DIGIT) | NEMO_DIGIT)
+                + pynini.closure(pynutil.delete(endings), 0, 1)
+            )
+            @ cardinal_graph
             + pynutil.insert("\"")
         )
 
         two_digit_year = NEMO_DIGIT ** (2) @ (cardinal.single_digits_graph | cardinal_graph)
         two_digit_year = pynutil.insert("year: \"") + two_digit_year + pynutil.insert("\"")
-
-        year_graph = pynutil.insert("year: \"") + year_graph + pynutil.insert("\"")
-        graph_year = pynini.closure(pynutil.delete(","), 0, 1) + delete_extra_space + year_graph
+        graph_year = (
+            ((pynutil.insert(" year: \"") + pynini.accep(",")) | pynutil.insert(" year: \""))
+            + pynutil.delete(" ")
```

Comment:
so we want to keep the komma instead of deleting?
could you refactor it to only have one instance of "pynutil.insert(" year: \"")" in this line

Reply:
done